core/server/worker/src/main/java/alluxio/worker/block/BlockLockManager.java

```diff
@@ -83,6 +84,23 @@ public ClientRWLock createNewResource() {
    */
   public BlockLockManager() {}
 
+  /**
+   * Locks a block. Note that even if this block does not exist, a closable lock is still returned.
+   *
+   * If all {@link PropertyKey#WORKER_TIERED_STORE_BLOCK_LOCKS} are already in use and no lock has
+   * been allocated for the specified block, this method will need to wait until a lock can be
+   * acquired from the lock pool.
+   *
+   * @param sessionId the session id
+   * @param blockId the block id
+   * @param blockLockType {@link BlockLockType#READ} or {@link BlockLockType#WRITE}
+   * @return closable block lock
+   */
+  public Closeable acquireBlockLock(long sessionId, long blockId, BlockLockType blockLockType) {
+    long lockId = lockBlock(sessionId, blockId, blockLockType);
+    return () -> unlockBlock(lockId);
```

Comment:
Can we wrap the unlocking code in a `CloseableResource` which has a built-in leak detector?

Reply:
Can have a private `BlockLock` that is a  subclass of `CloseableResource`, and make `pinBlock` return, and `unpinBlock` accept this `BlockLock`.

Reply:
Yes, I will implement the private block lock as you said if this interface looks good to Bin.

Reply:
yeah, that looks better. Agree with @dbw9580

Reply:
ClosableResource based lock implemented